llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2833,19 +2833,27 @@ static bool isNonZeroAdd(const APInt &DemandedElts, const SimplifyQuery &Q,
   if (matchOpWithOpEqZero(X, Y))
     return true;
 
-  if (NUW)
+  if (NUW) {
+    // X + Y != 0 if X != Y, and X + Y is NUW.
```

Comment:
A simpler way to state this:
```suggestion
    // If X != Y then X +nuw Y != 0.
```